bigchaindb/__init__.py

```diff
@@ -14,13 +28,7 @@
         'workers': None,  # if none, the value will be cpu_count * 2 + 1
         'threads': None,  # if none, the value will be cpu_count * 2 + 1
     },
-    'database': {
-        'backend': os.environ.get('BIGCHAINDB_DATABASE_BACKEND', 'rethinkdb'),
-        'host': os.environ.get('BIGCHAINDB_DATABASE_HOST', 'localhost'),
-        'port': int(os.environ.get('BIGCHAINDB_DATABASE_PORT', 28015)),
-        'name': os.environ.get('BIGCHAINDB_DATABASE_NAME', 'bigchain'),
-        'replicaset': os.environ.get('BIGCHAINDB_DATABASE_REPLICASET', 'bigchain-rs'),
-    },
+    'database': _database_rethinkdb,
```

Comment:
Rather than always default to the rethinkdb config, could we automatically use the setting that was given with `BIGCHAINDB_DATABASE_BACKEND`?

I guess something like:

```python
_database_map = {
    'mongodb': _database_mongodb,
    'rethinkdb': _database_rethinkdb,
}

config = {
    ...
   'database': _database_map[os.environ.get('BIGCHAINDB_DATABASE_BACKEND', 'rethinkdb')],
    ...
}
```

It might not be necessary because `autoconfigure()` should handle setting this properly, but from a glance, it does seem weird that `bigchaindb.config.database` might not reflect the setting of a running system.